Content.Server/GameObjects/Components/Observer/GhostComponent.cs

```diff
@@ -62,29 +76,82 @@ public override void HandleMessage(ComponentMessage message, IComponent componen
             base.HandleNetworkMessage(message, netChannel, session);
 
             switch (message)
-            {
-                case ReturnToBodyComponentMessage reenter:
-                    if (!Owner.TryGetComponent(out IActorComponent actor) || !CanReturnToBody) break;
-                    if (netChannel == null || netChannel == actor.playerSession.ConnectedClient)
-                    {
-                        actor.playerSession.ContentData().Mind.UnVisit();
-                        Owner.Delete();
-                    }
+                {
+                    case ReturnToBodyComponentMessage reenter:
+                        if (!Owner.TryGetComponent(out IActorComponent actor) || !CanReturnToBody) break;
+                        if (netChannel == null || netChannel == actor.playerSession.ConnectedClient)
+                        {
+                            actor.playerSession.ContentData().Mind.UnVisit();
+                            Owner.Delete();
+                        }
 
-                    break;
-                case ReturnToCloneComponentMessage reenter:
+                        break;
+                    case ReturnToCloneComponentMessage reenter:
 
-                    if (Owner.TryGetComponent(out VisitingMindComponent mind))
-                    {
-                        Owner.EntityManager.EventBus.RaiseEvent(EventSource.Local, new GhostReturnMessage(mind.Mind));
-                    }
+                        if (Owner.TryGetComponent(out VisitingMindComponent mind))
+                        {
+                            Owner.EntityManager.EventBus.RaiseEvent(EventSource.Local, new GhostReturnMessage(mind.Mind));
+                        }
 
-                    break;
-                default:
-                    break;
-            }
+                        break;
+
+                    case GhostWarpRequestMessage warp:
+                        if (warp.PlayerTarget != default)
+                        {
+                            foreach (var player in _playerManager.GetAllPlayers())
```

Comment:
idk if we want to have also ghost players

Reply:
Sorry for long delay, I think it would be fine for ghosts to teleport to other ghosts, imagine if one was talking about something at a certain location they're at other ghosts would likely want to see as well.